### PR TITLE
Many `From` implementations are undocumented #51430

### PR DESCRIPTION
**Pull request by sehinde-gb**
Hi, @skade This is my first pull request (Please be gentle with me....). I have followed your instructions, there could be a few mistakes. Let me know what they are and I will improve upon them.

### CODE REVIEW

**Comment by clarcharr** [comment]
I appreciate the effort to document these conversions! I think that right now the way rustdoc works, it's kind of hard to actually see the documentation for these. The docs for `From` are [extremely long](https://doc.rust-lang.org/nightly/std/convert/trait.From.html) and most of the impls are drowned out by the conversions between integers.

I think that adding docs on the impls themselves is a good thing, although maybe it might be better for some of these to put them elsewhere so they're also seen. For example, the impl for `Future` could go in the docs for `Future`, and the docs for converting between `&str` and `Box<str>` could go on `str`'s docs. Some of these end up just calling existing methods and could go alongside the docs for those methods.

Actually figuring out how to make these docs easy to find is a difficult problem, so, I think that modifying what you've got so far is definitely fine for now. Congrats on your first request. :)

**Comment by skade** [comment]
Hi @clarcharr, I think https://github.com/rust-lang/rust/issues/51430 is the better place for that kind of feedback.

Rustdoc drowning From implementations is an issue but IMHO not in scope for this.

**Comment by pietroalbini** [comment]
Thanks for this PR @ormrepo! The build is failing, and there are conflicts that needs to be solved. Do you have time to fix those?

**Comment by ormrepo** [comment]
Hi Pietro,

Sorry to hear about that. I submitted this a few weeks ago and I was hoping
that @skade <https://github.com/skade> would have responded to this by now.
Unfortunately he has disappeared for some unknown reason.

As this was my first PR I was hoping that my mentor @skade
<https://github.com/skade> would have assisted with this he may be away on
holiday or something.

Sorry about this.

Sehinde


On 9 July 2018 at 11:55:40, Pietro Albini (notifications@github.com) wrote:

Thanks for this PR @ormrepo <https://github.com/ormrepo>! The build is
failing, and there are conflicts that needs to be solved. Do you have time
to fix those?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/rust-lang/rust/pull/51738#issuecomment-403440016>, or mute
the thread

**Comment by pietroalbini** [comment]
@ormrepo don't worry, I'm just checking in to make sure PRs are not abandoned `:)`

**Comment by TimNN** [comment]
Ping from triage, @skade! This PR is waiting for your review! Also cc @rust-lang/docs in case @skade is unavailable in the near future.

**Comment by pietroalbini** [comment]
Ping from triage @skade @rust-lang/docs, this PR needs your review.

**Comment by pietroalbini** [comment]
Ping from triage! This PR needs a review, can @skade or someone else from @rust-lang/docs review this?

**Comment by alercah** [comment]
Won't the conversions also show up on the types with the `From` impls, and not only on the trait page?

**Review by alercah — CHANGES_REQUESTED** [review]

File: src/liballoc/boxed.rs
Comment (steveklabnik) [reply src/liballoc/boxed.rs]: Agreed

File: src/liballoc/boxed.rs
Comment (steveklabnik) [reply src/liballoc/boxed.rs]: Agreed.

**Review by sehinde-gb — COMMENTED** [review]
Thanks Steve for the clarification. And I agree with you. I don't have permission to approve these changes. I am a newbie here. Can someone please approve these changes and close down the PR request ? Or is this PR request owned by @skade I haven't heard from him in nearly 6 weeks.......

**Comment by Mark-Simulacrum** [comment]
I'm assigning @steveklabnik as the reviewer here. I believe the next step is to rebase and fix the comments left by @alercah and @steveklabnik.

r? @steveklabnik

**Comment by TimNN** [comment]
Ping from triage, @ormrepo. It looks like some changes have been requested to your PR.

**Comment by ormrepo** [comment]
Hi Triage or Tim,

I agree with @steveklabnik

I don’t have write access to approve this PR and I have been waiting on
@skade to approve the PR.
I have tried on multiple occasions to contact @skade and I have been
unsuccessful.

I am new to PR’s what is the process for assigning a PR if that person
refuses to respond ?

Regards
Sehinde


On 14 August 2018 at 17:32:49, Tim Neumann (notifications@github.com) wrote:

Ping from triage, @ormrepo <https://github.com/ormrepo>. It looks like some
changes have been requested to your PR.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/rust-lang/rust/pull/51738#issuecomment-412934504>, or mute
the thread

**Comment by TimNN** [comment]
Thank you for contributing to the Rust project, @ormrepo. Please note that this Pull Request cannot be approved right now. Reviewers have requested some changes, which you need to make before approval can be granted. (Note that `@skade` does not necessarily have to be involved in the process if they are unreachable at the moment. Others, like `@steveklabnik`, can also approve this PR once the requested changes have been made).

---

To make the requested changes, you'll have to go to your local checkout of the `boxrs_edits` branch, make the requested changes, and then finally commit and push them to GitHub.

**Comment by ormrepo** [comment]
Hi Tim,

I have made the changes on my branch and I am having some problems pushing
them up. I don't have an upstream branch specified, this means that I can’t
pull down the latest updates in order to rebase. I am at a loss right now
on how to move on. Do you have any ideas ?



Regards
Sehinde


On 15 August 2018 at 10:13:17, Tim Neumann (notifications@github.com) wrote:

Thank you for contributing to the Rust project, @ormrepo
<https://github.com/ormrepo>. Please note that this Pull Request cannot be
approved right now. Reviewers have requested some changes, which you need
to make before approval can be granted. (Note that @skade does not
necessarily have to be involved in the process if they are unreachable at
the moment. Others, like @steveklabnik, can also approve this PR once the
requested changes have been made).
------------------------------

To make the requested changes, you'll have to go to your local checkout of
the boxrs_edits branch, make the requested changes, and then finally commit
and push them to GitHub.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/rust-lang/rust/pull/51738#issuecomment-413140272>, or mute
the thread

**Comment by pietroalbini** [comment]
Ping from triage @ormrepo! It's been a while since we heard from you, will you have time to work on this again?

**Pull request closed without merging**
